src/api-client.ts

```diff
@@ -14,10 +14,12 @@ export class ApiClient {
             baseURL: this.apiBaseUrl,
             proxy: this.options?.proxy,
             timeout: this.options?.timeoutInMs,
-            headers: {
-                "X-API-Key": this.authProvider.getApiKey(),
-                "User-Agent": this.getUserAgent()
-            }
+        });
+
+        this.axiosInstance.interceptors.request.use(async (config: any) => {
+            config.headers.common["X-API-Key"] = await this.authProvider.getApiKey();
+            config.headers.common["User-Agent"] = this.getUserAgent();
+            return config;
```

Comment:
Why should we run this on every request? These header don't change. But if we already have an interceptor, let's move `signJwt()` into it

Reply:
Where do you suggest calculating it if not here? It's an async function, can't execute it in the constructor

Reply:
I agree. 
We should move sign_jwt() in here as well. 
This way the consumer can use the same SDK instance to run commands on multiple accounts. Because the API Key and the JWT are coupled it makes little sense to allow changing only one.